platform/linux-generic/m4/configure.m4

```diff
@@ -3,7 +3,10 @@ ODP_IMPLEMENTATION_NAME="odp-linux"
 ODP_VISIBILITY
 ODP_ATOMIC
 
-ODP_PTHREAD
+ODP_CHECK_CFLAG([-pthread])
+PTHREAD_LIBS_LIBODP="-lpthread"
+AC_SUBST([PTHREAD_LIBS_LIBODP])
```

Comment:
1) This can become implementation of `ODP_PTHREAD`, thus simplifying porting this to ODP-DPDK
2) Linker option is `-pthread`, not `-lpthread`, as far as I remember. And availability of that option (or library) also should be checked.